component/vdr/orb/vdr.go

```diff
@@ -648,12 +648,22 @@ func (v *VDR) Update(didDoc *docdid.Doc, opts ...vdrapi.DIDMethodOption) error {
 	updateOpt = append(updateOpt, updatedAlsoKnownAsOpts...)
 
 	updateOpt = append(updateOpt, update.WithSidetreeEndpoint(func() ([]string, error) {
-		var endpoint *models.Endpoint
+		endpoint, errGet := v.discoveryService.GetEndpoint(docResolution.DocumentMetadata.Method.AnchorOrigin)
+		if errGet != nil {
+			logger.Warnf("failed to get anchor origin %s endpoints will choose random domain: %w",
+				docResolution.DocumentMetadata.Method.AnchorOrigin, errGet)
 
-		// TODO make sure it's latest anchor origin
-		endpoint, err = v.discoveryService.GetEndpoint(docResolution.DocumentMetadata.Method.AnchorOrigin)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get endpoints: %w", err)
+			domain, errChoose := v.selectDomainSvc.Choose(v.domains)
```

Comment:
Is it possible that the same domain will be chosen?

Reply:
Yes. I will add retry

Reply:
done